setuptools/build_meta_legacy.py

```diff
@@ -0,0 +1,44 @@
+"""Compatibility backend for setuptools
+
+This is a version of setuptools.build_meta that endeavors to maintain backwards
+compatibility with pre-PEP 517 modes of invocation. It exists as a temporary
+bridge between the old packaging mechanism and the new packaging mechanism,
+and will eventually be removed.
+"""
+
+import sys
+
+from setuptools.build_meta import _BuildMetaBackend
+from setuptools.build_meta import SetupRequirementsError
+
+
+__all__ = ['get_requires_for_build_sdist',
+           'get_requires_for_build_wheel',
+           'prepare_metadata_for_build_wheel',
+           'build_wheel',
+           'build_sdist',
+           'SetupRequirementsError']
+
+
+class _BuildMetaLegacyBackend(_BuildMetaBackend):
+    def run_setup(self, setup_script='setup.py'):
```

Comment:
The frontend should be responsible for guaranteeing the backend is run with CWD set to the source root IMO (otherwise there’s no way the backend knows where to chdir to).

Reply:
It doesn't need to change the CWD (since Python doesn't do that for direct invocation), but the path entry to add is `os.path.abspath(setup_script)` (since that's what direct invocation would do).

Reply:
Just catching up here - the reason I didn't `cwd` is because from what @uranusjr said, the frontend does that for you, plus the original `python setup.py` execution *does not* do that.

I did not realize that direct invocation adds the dirname of `__file__` to the path, so yeah I can fix that.

Reply:
Indeed, PEP 517 guarantees that the hooks are called with the root of the source tree in the CWD:

> All hooks are run with working directory set to the root of the source tree,

(I don't know if there's any corner case where `setup.py` could be somewhere other than the root directory.)